### ISSUE TITLE
Better document different storage engine trade-offs

### ISSUE THREAD

**Issue by mbroecheler** [thinkaurelius/titan#891]
with respect to consistency, locking, caching, etc

**Comment by donaldquixote** [comment]
Yes please. Is there any resource that goes more in depth on the effects of the various configurations for Titan's TransactionBuilder wrt a Cassandra backend? 

In general, I understand the implications of using Cassandra, and understanding the Cassandra data model helps to mitigate some consistency concerns, but I don't understand exactly how a Titan graph maps to Cassandra column families, so it is difficult to determine exactly what a Titan transaction means on a Cassandra backend.
